### PR TITLE
useState translation

### PR DESCRIPTION
**Pull request by r-m169**

### CODE REVIEW

**Comment by r-m169** [comment]
I hope my request will be approved

File: src/content/reference/react/useState.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ title: useState
 
 <Intro>
 
-`useState` is a React Hook that lets you add a [state variable](/learn/state-a-components-memory) to your component.
+ `useState` هو Hook في React يتيح لك إضافة متغير حالة إلى مكونك
```
Comment (A7med3bdulBaset) [inline src/content/reference/react/useState.md:7]: في الصفحات الأخرى تم ترجمة كلمة hook إلى خطاف. يفضل أن تترجمها للتلائم مع باقي الصفحات

File: src/content/reference/react/useState.md [hunk 2]
```diff
@@ -16,11 +16,12 @@ const [state, setState] = useState(initialState);
 
 ---
 
-## Reference {/*reference*/}
+## الإشارة الى المرجع {/*reference*/}
```
Comment (A7med3bdulBaset) [inline src/content/reference/react/useState.md:19]: تكفي كلمة "المرجع"

File: src/content/reference/react/useState.md [hunk 3]
```diff
@@ -32,33 +33,32 @@ function MyComponent() {
   // ...
 ```
 
-The convention is to name state variables like `[something, setSomething]` using [array destructuring.](https://javascript.info/destructuring-assignment)
+التقليد هو تسمية متغيرات الحالة بشكل مثل `[something, setSomething]` باستخدام [array destructuring] [تفكيك الصفائف](https://javascript.info/destructuring-assignment)
```
Comment (A7med3bdulBaset) [inline src/content/reference/react/useState.md:36]: اعتدنا في الترجمة أني نترجم array إلى مصفوفة

**Comment by A7med3bdulBaset** [comment]
I'll complete reviewing later. 
I pushed some commits please review them.

**Please** read your translation again and rephrase what require

**Comment by A7med3bdulBaset** [comment]
Here's the corrected version of your text with grammar and wording improvements:

**Short**: That happened because you rebased between commits.

**TL;DR**:
First, when you started translating, you edited the main branch. Then, you rebased the branch with the main repository using `git pull`.

If you're collaborating on a project with other developers (like this one), refrain from making direct edits to the main branch. Instead, create a new branch if you need to make edits:
```sh
# Create a new branch
git branch translating-use-state

# Switch to the new branch
git checkout translating-use-state

# Alternatively, you can combine the two previous commands into one
git checkout -b translating-use-state
# This will create a branch named "translating-use-state" and switch to it.
```
Make the necessary edits, and then commit them, as usual, using `git add .` and `git commit -m "message"`. Afterward, push them using:
```sh
# When pushing the branch for the first time
git push -u origin translating-use-state
# After the first push
git push
```
This approach keeps your main branch untouched.

When you intend to start a new branch:
```sh
# Return to the main branch
git checkout main
# Create a new branch
git checkout -b translating-use-effect
```

I have a solution for your issue, but it might be complex and challenging for you. Therefore, consider deleting your forked repository, as you made edits directly to the main branch. Then, fork the repository again. In a separate directory on your computer, clone the repository again and create a new branch as instructed earlier. Copy your translation [from this link](https://gist.github.com/A7med3bdulBaset/a0a6e94cce8455056058165f99e61995) and paste it at the end of this comment. Finally, open a new pull request
```sh
# Execute these commands one by one
git clone https://github.com/r-m169/ar.react.dev.git

# You'll now have a new directory named 'ar.react.dev'
cd ar.react.dev

# Open Visual Studio Code
code .

# Create a new branch and switch to it
git checkout -b translating-use-state

# Now copy the text from the link and paste it into 'src/content/reference/react/useState.md'
# Make your changes

# Then push the branch
git push -u origin translating-use-state

# Commit your changes
git add .
git commit -m "Translating useState reference"
git push
```

[Your translation can be found here](https://gist.github.com/A7med3bdulBaset/a0a6e94cce8455056058165f99e61995).

Let me know if you face a problem.
